htsget-config/src/storage/url.rs

```diff
@@ -1,45 +1,50 @@
 use std::str::FromStr;
 
+use http::Uri as InnerUrl;
 use serde::{Deserialize, Serialize};
-use url::Url as InnerUrl;
 
 use crate::error::Error::ParseError;
 use crate::error::{Error, Result};
 use crate::storage::local::default_authority;
 use crate::types::Scheme;
 
-fn default_url() -> Url {
-  Url(InnerUrl::from_str(&format!("https://{}", default_authority())).expect("expected valid url"))
+fn default_url() -> ValidatedUrl {
+  ValidatedUrl(Url(
+    InnerUrl::from_str(&format!("https://{}", default_authority())).expect("expected valid url"),
+  ))
 }
 
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
 #[serde(default)]
 pub struct UrlStorage {
-  url: Url,
+  url: ValidatedUrl,
   response_scheme: Scheme,
   forward_headers: bool,
 }
 
-/// A new type struct on top of `url::Url` which only allows http or https schemes when deserializing.
+/// A wrapper around `http::Uri` type which implements serialize and deserialize.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
-#[serde(try_from = "InnerUrl")]
-pub struct Url(InnerUrl);
+struct Url(#[serde(with = "http_serde::uri")] InnerUrl);
 
-impl Url {
+/// A new type struct on top of `http::Uri` which only allows http or https schemes when deserializing.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
+#[serde(try_from = "Url")]
+pub struct ValidatedUrl(Url);
+
+impl ValidatedUrl {
   /// Get the inner url.
   pub fn into_inner(self) -> InnerUrl {
-    self.0
+    self.0 .0
```

Comment:
Woops, can we have those fields with names instead of `.0`?

Reply:
We could, although a new type is convenient because it is serialised and deserialized directly as the inner type. I thought this approach was probably the best even though it's uglier, as it's just a private inner type anyway (I.e. `Url` is not pub exported). I'll have a look to see if there's a way to serialise/deserialise a regular struct directly as an inner field though.

Reply:
Ideally, there wouldn't be two `Url` types here in the first place, but `http_serde::uri` wasn't working when I swapped `Url` for `Uri`.